build/scripts/tests.sh

```diff
@@ -20,43 +20,38 @@ TARGET_FRAMEWORK=netcoreapp2.0
 XUNIT_CONSOLE_VERSION=$(get_package_version dotnet-xunit)
 XUNIT_CONSOLE=${NUGET_DIR}/dotnet-xunit/${XUNIT_CONSOLE_VERSION}/tools/${TARGET_FRAMEWORK}/xunit.console.dll
 
-echo Using $XUNIT_CONSOLE
+echo "Using ${XUNIT_CONSOLE}"
 
 # Need to publish projects that have runtime assets before running tests
 NEED_PUBLISH=(
-    'src\Compilers\CSharp\Test\Symbol\CSharpCompilerSymbolTest.csproj'
+    'src/Compilers/CSharp/Test/Symbol/CSharpCompilerSymbolTest.csproj'
 )
-for p in $NEED_PUBLISH
+
+for project in $NEED_PUBLISH
 do
-    echo Publishing ${p}
-    dotnet publish --no-restore ${SRC_PATH}/${p} -p:RuntimeIdentifier=${RUNTIME_ID} -p:TargetFramework=${TARGET_FRAMEWORK} -p:SelfContained=true
+    echo "Publishing ${project}"
+    dotnet publish --no-restore ${ROOT_PATH}/${project} -r ${RUNTIME_ID} -f ${TARGET_FRAMEWORK} -p:SelfContained=true
 done
 
 # Discover and run the tests
-mkdir -p ${LOG_DIR}
-pushd ${UNITTEST_DIR}
+mkdir -p "${LOG_DIR}"
 
-for d in *
+for TEST_PATH in ${UNITTEST_DIR}/*/${TARGET_FRAMEWORK}
 do
-    TEST_PATH=${UNITTEST_DIR}/${d}/${TARGET_FRAMEWORK}
     PUBLISH_TEST_PATH=${TEST_PATH}/${RUNTIME_ID}/publish
     if [ -d ${PUBLISH_TEST_PATH} ]
     then
         TEST_PATH=${PUBLISH_TEST_PATH}
     fi
 
-    pushd $TEST_PATH
-    FILE_NAME=$(ls *.UnitTests.dll)
-    LOG_NAME="${FILE_NAME%.*}.xml"
-    echo Running ${TEST_PATH}/${FILE_NAME}
-    dotnet exec --depsfile "${FILE_NAME%.*}.deps.json" --runtimeconfig "${FILE_NAME%.*}.runtimeconfig.json" ${XUNIT_CONSOLE} $FILE_NAME -xml ${LOG_DIR}/${LOG_NAME}
-    if [ $? -ne 0 ]; then
+    FILE_NAME=( ${TEST_PATH}/*.UnitTests.dll )
```

Comment:
what is the significance of the extra parens here?

Reply:
Parens creates an array, and expands the glob. If this was `FILE_NAME=${TEST_PATH}/*.UnitTests.dll` (same thing, minus parens), then the glob would not be expanded. We then assume the array contains one element (because `${var}` where var is an array is implicitly `${var[0]}`), like we were doing before.

(I believe this was what you were trying to work around with `$(ls the*glob)`)

Reply:
neat. is that doc anywhere that i can read further on?

Reply:
Docs? What docs. I don't know these "docs" that you're speaking of 😅

(My knowledge here came from [stackoverflow](https://unix.stackexchange.com/questions/34011/the-best-way-to-expand-glob-pattern) [answers](https://stackoverflow.com/questions/9229870/assign-results-of-globbing-to-a-variable-in-bash) [like](https://unix.stackexchange.com/questions/314804/how-to-make-bash-glob-a-string-variable) [these](https://stackoverflow.com/questions/369145/how-to-assign-a-glob-expression-to-a-variable-in-a-bash-script) )